src/redux/selectors/userSelectors.js

```diff
@@ -0,0 +1 @@
+export const getHasUser = (state) => state.users && Object.keys(state.users.userInfo).length > 0;
```

Comment:
this is not how you create a selector.

it should at least pass through `reselect`'s `createSelector` function to use its memoized feature

```js
import { createSelector } from 'reselect'

const loggedUser = (state) => state.users; 

export const getHasUser = createSelector([loggedUser], (user) => user && Object.keys(user.userInfo).length > 0); 
```

Reply:
please confirm that this^ works. I just typed it via github